generator/src/lib.rs

```diff
@@ -1583,16 +1583,20 @@ pub fn derive_setters(
 
     let extends_name = name_to_tokens(&format!("Extends{}", name));
 
-    let next_extends = _struct.extends.as_ref().map(|extends| {
-        let extends = extends
-            .split(',')
-            .find(|extend| root_create_info_names.contains(&extend.to_string()))
-            .expect("Should have a root create info");
-        name_to_tokens(&format!("Extends{}", name_to_tokens(&extends)))
-    });
+    let root_extends: Vec<Ident> = _struct
+        .extends
+        .as_ref()
+        .map(|extends| {
+            extends
+                .split(',')
+                .filter(|extend| root_create_info_names.contains(&extend.to_string()))
+                .map(|extends| name_to_tokens(&format!("Extends{}", name_to_tokens(&extends))))
+                .collect()
+        })
+        .unwrap_or(vec![]);
 
     // We only implement a next methods for root create infos
-    let next_function = if has_next && next_extends.is_none() {
+    let next_function = if has_next && root_extends.is_empty() {
         quote! {
             /// Prepends the given extension struct between the root and the first pointer. This
             /// method only exists on create infos that can be passed to a function directly. Only
```

Comment:
> This method only exists on create infos that can be passed to a function directly

This is incorrect, because this method exists (and is useful) on output structs too.

Reply:
Do you have an example?

Reply:
For example, it should be (but seemingly is not yet) defined for `VkPhysicalDeviceProperties2`. The same goes for any other root output struct with a `p_next` member.

Reply:
The current implementation doesn't allow for pointer chains to be added. As in `push_next` on `VkPhysicalDeviceProperties2` and then insert `VkPhysicalDeviceProperties2` into `DeviceCreateInfo`. We would need to iterate over the chains (which is fine). I don't think this is necessary though. `push_next` only on root create infos seems fine to me.

Reply:
I don't understand what you're trying to say. `VkPhysicalDeviceProperties2` is an output struct; it does not extend `VkDeviceCreateInfo`, it is used to return data from `vkGetPhysicalDeviceProperties2`. It, and every struct like it, needs exactly the same extension semantics as you've implemented so far for root `CreateInfo` structs.

Reply:
Okay I see what you mean. It is correctly implemented (as in `VkPhysicalDeviceProperties2` has a `push_next` fn).  I'll just have to update the comments.

Reply:
Oh, so it is, was just searching badly.